shader/flatten.wgsl

```diff
@@ -227,7 +225,59 @@ fn flatten_cubic(cubic: Cubic) {
     }
 }
 
-var<private> pathdata_base: u32;
+fn draw_join(
+    stroke: vec2f, path_ix: u32, style_flags: u32, p0: vec2f,
+    tan_prev: vec2f, tan_next: vec2f,
+    n_prev: vec2f, n_next: vec2f,
+    transform: Transform,
+) {
+    switch style_flags & STYLE_FLAGS_JOIN_MASK {
+        case /*STYLE_FLAGS_JOIN_BITS_BEVEL*/0u: {
+            output_two_lines_with_transform(path_ix,
+                                            p0 + n_prev, p0 + n_next,
+                                            p0 - n_next, p0 - n_prev,
+                                            transform);
+        }
+        case /*STYLE_FLAGS_JOIN_BITS_MITER*/0x10000000u: {
+            let c = tan_prev.x * tan_next.y - tan_prev.y * tan_next.x;
+            let d = dot(tan_prev, tan_next);
+            let hypot = length(vec2f(c, d));
+            let miter_limit = unpack2x16float(style_flags & STYLE_MITER_LIMIT_MASK).x;
```

Comment:
I'd *probably* choose to use the indexing form of vector access (i.e. `unpack2x16float(...)[0]`) here, as this isn't semantically an `x` point but just the first 16 bits.

But that's not based on any checking what the style is elsewhere

Reply:
We have some uses of the `unpack` intrinsics in the fine shader that perform a swizzle using the `xyzw` accessors, though I suppose changing those to use the color (`rgba`) accessors would be semantically more correct.

Using `[0]` here sounds good to me.